src/peft/peft_model.py

```diff
@@ -1604,6 +1604,9 @@ def forward(
 
         if peft_config.peft_type == PeftType.PREFIX_TUNING:
             past_key_values = self.get_prompt(batch_size)
+            # check if past_kv already in kwargs
+            if "past_key_values" in kwargs:
+                kwargs.pop("past_key_values")
```

Comment:
I guess we could also do: 

```python
kwargs["past_key_values"] = self.get_prompt(batch_size)
```

and below remove the `past_key_values=past_key_values` from the call. WDYT is better?

Also, I wonder if we should check if the existing `past_key_values` is `None`? I think that's the case for Llava but I wonder if we can just silently drop it if it's not `None`.

Reply:
I think in the context of prefix-tuning, the `past_key_values` should be overwritten no matter it's `None` or not. Otherwise the virtual prompts are excluded from the forward pass in deeper layers.

Reply:
All done :)